Generated code WinRT.IActivationFactoryMethods missing global

**Describe the bug**
Generated code `WinRT.IActivationFactoryMethods` missing `global::`, which course `error CS0234: 命名空间“FullTrustUWP.WinRT”中不存在类型或命名空间名“IActivationFactoryMethods”(是否缺少程序集引用?)`.

![image](https://github.com/microsoft/CsWinRT/assets/27689196/882106ee-67fd-4633-9309-c71607c163ad)

**To Reproduce**
1. Create a winrt class which namespace ending with `.WinRT`
2. Reference it by a C#/WinRT project

**Expected behavior**
Generate `global::WinRT.IActivationFactoryMethods`.

**Version Info**
cswinrt.exe 2.0.6.240201.1

--- a/src/Tests/TestComponentCSharp/TestComponentCSharp.idl
+++ b/src/Tests/TestComponentCSharp/TestComponentCSharp.idl
@@ -650,6 +650,18 @@ And this is another one"
         };
     }
 
+    // Compile time test for sub WinRT namespace
+    namespace WinRT
+    {
+        runtimeclass Class
+        {
+            Class();
+            static void StaticMethod();
+            void Method();
+            void Method2(Windows.Foundation.IStringable stringable);
+        }
+    }
+
     namespace AnotherAssembly
     {
         [default_interface]

--- a/src/Tests/TestComponentCSharp/TestComponentCSharp.vcxproj
+++ b/src/Tests/TestComponentCSharp/TestComponentCSharp.vcxproj
@@ -94,6 +94,7 @@
     <ClInclude Include="WarningClass.h" />
     <ClInclude Include="WarningStatic.h" />
     <ClInclude Include="Windows.Class.h" />
+    <ClInclude Include="WinRT.Class.h" />
   </ItemGroup>
   <ItemGroup>
     <ClCompile Include="$(GeneratedFilesDir)module.g.cpp" />
@@ -114,6 +115,7 @@
     <ClCompile Include="WarningClass.cpp" />
     <ClCompile Include="WarningStatic.cpp" />
     <ClCompile Include="Windows.Class.cpp" />
+    <ClCompile Include="WinRT.Class.cpp" />
   </ItemGroup>
   <ItemGroup>
     <Midl Include="TestComponentCSharp.idl" />

--- a/src/Tests/TestComponentCSharp/WinRT.Class.cpp
+++ b/src/Tests/TestComponentCSharp/WinRT.Class.cpp
@@ -1,0 +1,18 @@
+#include "pch.h"
+#include "WinRT.Class.h"
+#include "WinRT.Class.g.cpp"
+
+namespace winrt::TestComponentCSharp::WinRT::implementation
+{
+    void Class::StaticMethod()
+    {
+    }
+
+    void Class::Method()
+    {
+    }
+
+    void Class::Method2(winrt::Windows::Foundation::IStringable const& stringable)
+    {
+    }
+}

--- a/src/Tests/TestComponentCSharp/WinRT.Class.h
+++ b/src/Tests/TestComponentCSharp/WinRT.Class.h
@@ -1,0 +1,20 @@
+#pragma once
+#include "WinRT.Class.g.h"
+
+namespace winrt::TestComponentCSharp::WinRT::implementation
+{
+    struct Class : ClassT<Class>
+    {
+        Class() = default;
+
+        static void StaticMethod();
+        void Method();
+        void Method2(winrt::Windows::Foundation::IStringable const& stringable);
+    };
+}
+namespace winrt::TestComponentCSharp::WinRT::factory_implementation
+{
+    struct Class : ClassT<Class, implementation::Class>
+    {
+    };
+}

--- a/src/cswinrt/code_writers.h
+++ b/src/cswinrt/code_writers.h
@@ -2074,7 +2074,7 @@ ComWrappersSupport.RegisterObjectForInterface(this, ThisPtr);
             auto objrefname = w.write_temp("%", bind<write_objref_type_name>(class_type));
 
             w.write(R"(
-public %() : this(%(WinRT.IActivationFactoryMethods.ActivateInstance<IUnknownVftbl>(%)))
+public %() : this(%(global::WinRT.IActivationFactoryMethods.ActivateInstance<IUnknownVftbl>(%)))
 {
 ComWrappersSupport.RegisterObjectForInterface(this, ThisPtr);
 %

